From eba7340617b13bb4158f62681b36914180477c9a Mon Sep 17 00:00:00 2001
From: Pengxuan Zheng <pzheng@quicinc.com>
Date: Thu, 10 Apr 2025 13:56:52 -0700
Subject: [PATCH] Revert "[ARM][ConstantIslands] Correct MinNoSplitDisp
 calculation (#114590)"

This reverts commit e48916f615e0ad2b994b2b785d4fe1b8a98bc322.

patch.cherry: true
patch.metadata.original_sha: 36acaa0be5cf6273200d4cc4c35042c4bfc3861b
patch.platforms: chromiumos
patch.version_range.from: 559379
patch.version_range.until: 571637

BUG=b:409992109
TEST=None

---
 llvm/lib/Target/ARM/ARMConstantIslandPass.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/llvm/lib/Target/ARM/ARMConstantIslandPass.cpp b/llvm/lib/Target/ARM/ARMConstantIslandPass.cpp
index e41e02a560db..89eb49ed416a 100644
--- a/llvm/lib/Target/ARM/ARMConstantIslandPass.cpp
+++ b/llvm/lib/Target/ARM/ARMConstantIslandPass.cpp
@@ -1323,8 +1323,7 @@ bool ARMConstantIslands::findAvailableWater(CPUser &U, unsigned UserOffset,
   MachineBasicBlock *UserBB = U.MI->getParent();
   BBInfoVector &BBInfo = BBUtils->getBBInfo();
   const Align CPEAlign = getCPEAlign(U.CPEMI);
-  unsigned MinNoSplitDisp =
-      BBInfo[UserBB->getNumber()].postOffset(CPEAlign) - UserOffset;
+  unsigned MinNoSplitDisp = BBInfo[UserBB->getNumber()].postOffset(CPEAlign);
   if (CloserWater && MinNoSplitDisp > U.getMaxDisp() / 2)
     return false;
   for (water_iterator IP = std::prev(WaterList.end()), B = WaterList.begin();;
-- 
2.49.0.1112.g889b7c5bd8-goog